libs/server/Storage/SizeTracker/CacheSizeTracker.cs

```diff
@@ -96,13 +95,8 @@ public void Start(CancellationToken token)
         /// <param name="newTargetSize">Target size</param>
         public (long mainLogSizeBytes, long readCacheSizeBytes) CalculateLogTargetSizeBytes(long newTargetSize)
         {
-            long residual = newTargetSize - IndexSizeBytes;
-
-            if (residual <= 0)
-                throw new TsavoriteException($"Target size {newTargetSize} must be larger than index size {IndexSizeBytes}");
-
-            var mainLogSizeBytes = this.store.ReadCache == null ? residual : residual / 2;
-            var readCacheSizeBytes = this.store.ReadCache == null ? 0 : residual / 2;
+            var mainLogSizeBytes = this.store.ReadCache == null ? newTargetSize : newTargetSize / 2;
```

Comment:
there is an implicit assumption that we split main log and read cache 50:50 here. this needs to be configured by user

Reply:
These could change in a separate PR that enables read cache in Garnet.